src/Toolkit/Str.php

```diff
@@ -264,17 +266,33 @@ public static function contains(string $string = null, string $needle, bool $cas
      * according to locale settings
      *
      * @param int|null $time
-     * @param string|null $format
-     * @param string $handler date or strftime
+     * @param string|array|\IntlDateFormatter|null $format
+     * @param string $handler date, intl or strftime
      * @return string|int
      */
-    public static function date(?int $time = null, ?string $format = null, string $handler = 'date')
+    public static function date(?int $time = null, $format = null, string $handler = 'date')
     {
         if (is_null($format) === true) {
             return $time;
         }
 
-        // separately handle strftime to be able
+        // $format is an IntlDateFormatter instance
+        if (is_a($format, 'IntlDateFormatter') === true) {
+            return $format->format($time);
+        }
+
+        // if $format is an array, `intl` needs to be (default) handler
+        if (is_array($format) === true) {
+            $handler = 'intl';
+        }
```

Comment:
To be honest I feel like we should get rid of the `array` format type / not add it in the first place:

- It's not needed as advanced users can also pass an `IntlDateFormatter` object
- The type hint for the argument becomes confusing (we would need to explain what the array means and that it's only for the `intl` handler)
- We would need checks that prevent `array` for the other two handlers. Otherwise you will get errors if you pass an array for `date()`.